pkg/defaults/defaults.go

```diff
@@ -373,6 +373,7 @@ func stepsForStepImages(
 			config := api.InputImageTagStepConfiguration{
 				BaseImage: *subStep.FromImage,
 				To:        link,
+				Source:    fmt.Sprintf(api.ImageStreamSourceTest, subStep.As),
```

Comment:
We use the `config` struct as map key to deduplicate, that will break now if different steps use the same `Source`

Reply:
Do you know where that happens so I can take a look and see about tweaking that deduplication?

Reply:
two lines below, the `if _, ok := inputImages[config]; ok { continue }`

The implication is also that there could be more than one Source I think

Reply:
Cool, thanks! Still getting used to Go's syntax :)

When you say that there could be more than one Source, how would that happen?

Reply:
Multiple steps that all need the same base image. The fact that we de-duplicate them seems to imply that is possible, but I am not super familiar with this part of the code base. @bbguimaraes might know more

Reply:
Yes, what Alvaro says is correct. We may get duplicates by e.g. multiple multi-stage steps using the same image in their `from_image` staza, or having the same thing in a `base_images` item and as a multi-stage step image, etc. 

Not sure if we validate this, but this could be a minimal reproducer if we don't:

```yaml
base_images:                                                                                                                                                                                                                                                                              
  base:
    name: "4.8"
    namespace: ocp
     tag: base
  all-your-bases:
    name: "4.8"
    namespace: ocp
    tag: base
```

Reply:
In reality this is most likely to happen when multiple steps need the same image

Reply:
Ah, I gotcha. Ok, that makes sense.

Reply:
To super-clarify, here's a step that uses an image: https://github.com/openshift/release/blob/master/ci-operator/step-registry/gather/loki/gather-loki-ref.yaml#L4-L7.

We'll have duplicate sources if there are more steps with the idential `from_image`, or if a this step is used from a config that importes the same image in `base_images`.